src/coprocessor/checksum.rs

```diff
@@ -30,62 +29,40 @@ pub struct ChecksumContext {
     store: SnapshotStore,
     ranges: IntoIter<KeyRange>,
     scanner: Option<Scanner>,
+    req_ctx: ReqContext,
+    metrics: ExecutorMetrics,
 }
 
 impl ChecksumContext {
     pub fn new(
         req: ChecksumRequest,
         ranges: Vec<KeyRange>,
         snap: Box<Snapshot>,
-        ctx: &ReqContext,
-    ) -> ChecksumContext {
+        req_ctx: ReqContext,
+    ) -> Result<ChecksumContext> {
         let store = SnapshotStore::new(
             snap,
             req.get_start_ts(),
-            ctx.isolation_level,
-            ctx.fill_cache,
+            req_ctx.context.get_isolation_level(),
+            !req_ctx.context.get_not_fill_cache(),
         );
-        ChecksumContext {
+        Ok(ChecksumContext {
```

Comment:
Why return result instead of `ChecksumContext`?

Reply:
To keep similar interface among these `RequestHandlers`. It will be wired for some of them return `Result<RequestHandlers>` in `new()` and others return `RequestHandlers`.